### ISSUE TITLE
Add Cop config to prevent inline disabling

### ISSUE THREAD

**Issue by caalberts** [rubocop/rubocop#9376]
## Is your feature request related to a problem? Please describe.

When introducing a new Cop, sometimes we may want to prevent inline disabling altogether. This is to prevent a pattern where a discouraged practice continues to happen with Rubocop disable comments sprinkled across the codebase.

## Describe the solution you'd like

A Cop configuration such as follows:
```yml
Style/For:
  Enabled: true
  AllowInlineDisable: false # default is true
```

When a code comment such as follows exist, it should `rubocop` should fail.

```rb
for x in (0..19) # rubocop:disable Style/For
```

## Describe alternatives you've considered

I'm not sure if this should be a setting per Cop, or maybe a setting for a RuboCop syntax linter if there's a such Cop.

**Comment by caalberts** [comment]
I would be happy to help implement this as well

**Comment by dvandersluis** [comment]
I like this idea, maybe with `Required: true`, or somehow making use of `Severity`. One prospective problem though is that what happens if a cop cannot be disabled that there isn't actually a fix for (a false negative perhaps, or a cop looking for a specific method name in the wrong context)?

**Comment by dvandersluis** [comment]
There already is `Style/DisableCopsWithinSourceCodeDirective` which does this. It either disallows all rubocop disable directives, or all except cops listed within `AllowedCops` configuration.

**Comment by aergonaut** [comment]
@dvandersluis I apologize for resurrecting a closed issue. If this is inappropriate, I will gladly create a new issue or discussion.

I think `Style/DisableCopsWithinSourceCodeDirective` is not enough to implement what @caalberts requested in the OP. While `Style/DisableCopsWithinSourceCodeDirective` does create an offense whenever `rubocop:disable` is used, because it is itself a Cop, `rubocop:disable Style/DisableCopsWithinSourceCodeDirective` can still turn it off. And as long as you disable it first, you will be able to disable other cops afterwards too.

So, instead of writing simply, e.g. `rubocop:disable Style/HashSyntax`, one could write instead `rubocop:disable Style/DisableCopsWithinSourceCodeDirective Style/HashSyntax`.

I think a `Required: true` is still needed. Or some way to control the directives, perhaps a config in `.rubocop.yml` like:

```yaml
Directives:
    disable: false
```
